sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSourceResolution.scala

```diff
@@ -188,18 +186,10 @@ case class DataSourceResolution(
       }
 
     case ShowTablesStatement(None, pattern) =>
-      defaultCatalog match {
-        case Some(catalog) =>
-          ShowTables(
-            catalog.asTableCatalog,
-            catalogManager.currentNamespace,
-            pattern)
-        case None =>
-          ShowTablesCommand(None, pattern)
-      }
+      ShowTables(currentCatalog.asTableCatalog, catalogManager.currentNamespace, pattern)
```

Comment:
If there is no catalog specified, this will not fallback to v1 anymore since `currentCatalog` always returns a `CatalogPlugin`. This should be OK right?

Reply:
I think so